### PR TITLE
Add QA with Langchain of AnalyticDB  vector store and OpenAI example.

### PR DESCRIPTION
**Pull request by wangxuqi**
Hi there,I hope you're doing well. 

I'm excited to open this PR. This is another example of using a fully Postgres syntax compatible database 'AnalyticDB' as a datastore to build QA with Langchain and OpenAI APIs. 

As AnalyticDB can be used with Langchain directly for now. This example is a better case for user who want to use  'AnalyticDB' as a vector store.

I would really appreciate your feedback.

### CODE REVIEW

**Comment by ted-at-openai** [comment]
Thanks, I'll take a look. What's your goal for this contribution?

**Comment by ted-at-openai** [comment]
Also, what kind of feedback are you looking for? If you think it's in good shape, I'll spend less time reviewing it before merging.

**Comment by wangxuqi** [comment]
> Also, what kind of feedback are you looking for? If you think it's in good shape, I'll spend less time reviewing it before merging.

So sorry if my commit message wasn't clear enough in conveying my intentions.  I think it's in good shape, This notebook presents how to implement a Question Answering system with LangChain, AnalyticDB and OpenAI embeddings. This example mainly consists of the following main steps.
- prepare connection string with Environment variables and LangChain
```
from langchain.vectorstores.analyticdb import AnalyticDB

CONNECTION_STRING = AnalyticDB.connection_string_from_db_params(
    driver=os.environ.get("PG_DRIVER", "psycopg2cffi"),
    host=os.environ.get("PG_HOST", "localhost"),
    port=int(os.environ.get("PG_PORT", "5432")),
    database=os.environ.get("PG_DATABASE", "postgres"),
    user=os.environ.get("PG_USER", "postgres"),
    password=os.environ.get("PG_PASSWORD", "postgres"),
)
```
- Definition a VectorStore Chain of AnalyticDB
```
from langchain.vectorstores import AnalyticDB
from langchain.embeddings import OpenAIEmbeddings
from langchain import VectorDBQA, OpenAI

embeddings = OpenAIEmbeddings()
doc_store = AnalyticDB.from_texts(
    texts=answers, embedding=embeddings, connection_string=CONNECTION_STRING,
    pre_delete_collection=True,
)
```
- Define QA chain with OpenAI llm and vector store defined above.
```
llm = OpenAI()
qa = VectorDBQA.from_chain_type(
    llm=llm,
    chain_type="stuff",
    vectorstore=doc_store,
    return_source_documents=False,
)
```
- SearchData with the QA chain
```
qa.run(question)
```

File: examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb
Comment (ted-at-openai) [inline examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:7]: Want to make this a markdown header?

File: examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb
Comment (ted-at-openai) [inline examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:8]: Optional: can add a markdown link to the other notebook

File: examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb
Comment (ted-at-openai) [inline examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:24]: Recommend making this a markdown bulleted list

File: examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb
Comment (ted-at-openai) [inline examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:34]: optional: explain what each of these is for, add links (I wasn't sure what psycopg2ffi was for and had to look it up)

File: examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb
Comment (ted-at-openai) [inline examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:89]: optional: new url is https://platform.openai.com/account/api-keys (platform instead of beta)

File: examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb
Comment (ted-at-openai) [inline examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:99]: "You" missing capitalization. ".And" missing space.

Comment (ted-at-openai) [reply examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:99]: final word missing period.

File: examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb
Comment (ted-at-openai) [inline examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:117]: "default" misspelled

File: examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb
Comment (ted-at-openai) [inline examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb:161]: Can be markdown header

**Review by ted-at-openai — CHANGES_REQUESTED** [review]
Looks great. I didn't run the whole thing myself, as I didn't want to go through the trouble of setting up a DB. I left some suggestions mostly on wording and formatting. Happy to merge once addressed. Thank you!

**Comment by wangxuqi** [comment]
> Looks great. I didn't run the whole thing myself, as I didn't want to go through the trouble of setting up a DB. I left some suggestions mostly on wording and formatting. Happy to merge once addressed. Thank you!
Thank you for your feedback! I've addressed your suggestions regarding wording and formatting. Thank you once again for your guidance and support, and I look forward to our continued collaboration.

**Comment by wangxuqi** [comment]
I also have a related PR in another project : https://github.com/openai/chatgpt-retrieval-plugin/pull/147. I would appreciate it if you could take some time to review it.

**Review by ted-at-openai — APPROVED** [review]
Didn't check carefully, but I trust you addressed the changes and are happy with the final result. Will merge.

**Pull request merged**
